tests/utils/skipif.php

```diff
@@ -97,7 +97,7 @@ function skip_if_not_auth_mechanism($authMechanism)
 {
     $uriAuthMechanism = get_uri_option(URI, 'authMechanism');
 
-    if ($uriAuthMechanism === null) {
+    if ($uriAuthMechanism === null && $authMechanism !== null) {
```

Comment:
This shouldn't be necessary, as `is_auth()` already returns `true` if `authMechanism` is set to something. Better to just use `skip_if_not_auth()`.

Also, if we were to keep this behavior, I'd suggest defaulting the argument to `null` and changing the doc block accordingly to mention that we check for "anything" if no argument is provided.

Reply:
This is necessary to make sure there **is** authentication, and it is *not* X509. ``skip_if_not_auth()`` returns true for either "standard", where ``authMechanism`` isn't set to anything, or ``MONGO-X509`` auth.

It's used at https://github.com/mongodb/mongo-php-driver/pull/907/files#diff-15f3c9a5beac9f38bb34f9eef5d058f3R6 to make sure we're just dealing with normal simple auth.